modules/wechat_qrcode/src/zxing/qrcode/decoder/decoded_bit_stream_parser.cpp

```diff
@@ -190,16 +191,20 @@ void DecodedBitStreamParser::decodeByteSegment(Ref<BitSource> bits_, string& res
                                                CharacterSetECI* currentCharacterSetECI,
                                                ArrayRef<ArrayRef<char> >& byteSegments,
                                                ErrorHandler& err_handler) {
-    int nBytes = count;
     BitSource& bits(*bits_);
     // Don't crash trying to read more bits than we have available.
     int available = bits.available();
     // try to repair count data if count data is invalid
     if (count * 8 > available) {
-        count = (available + 7 / 8);
+        count = (available + 7) / 8;
     }
+    size_t nBytes = count;
+
+    ArrayRef<char> bytes_(nBytes);
+    // issue https://github.com/opencv/opencv_contrib/issues/3478
+    if (bytes_->empty())
+        return;
```

Comment:
If I understand correctly, `bytes_->empty()` should be equivalent to `nBytes == 0`. So why not using `nBytes == 0` for simplicity and efficiency?

Reply:
I think this change would be a more visual indication of the intention to avoid null pointers?

Reply:
yes